cloud/blockstore/libs/service_local/file_io_service_provider_ut.cpp

```diff
@@ -0,0 +1,269 @@
+#include "file_io_service_provider.h"
+
+#include <cloud/storage/core/libs/common/file_io_service.h>
+
+#include <library/cpp/testing/unittest/registar.h>
+
+namespace NCloud::NBlockStore::NServer {
+
+namespace {
+
+////////////////////////////////////////////////////////////////////////////////
+
+struct TTestFileIOService
+    : IFileIOService
+{
+    int Started = 0;
+    int Stopped = 0;
+
+    void Start() override
+    {
+        ++Started;
+    }
+
+    void Stop() override
+    {
+        ++Stopped;
+    }
+
+    void AsyncRead(
+        TFileHandle& file,
+        i64 offset,
+        TArrayRef<char> buffer,
+        TFileIOCompletion* completion) override
+    {
+        Y_UNUSED(file);
+        Y_UNUSED(offset);
+        Y_UNUSED(buffer);
+        Y_UNUSED(completion);
+    }
+
+    void AsyncWrite(
+        TFileHandle& file,
+        i64 offset,
+        TArrayRef<const char> buffer,
+        TFileIOCompletion* completion) override
+    {
+        Y_UNUSED(file);
+        Y_UNUSED(offset);
+        Y_UNUSED(buffer);
+        Y_UNUSED(completion);
+    }
+};
+
+////////////////////////////////////////////////////////////////////////////////
+
+struct TTestProvider
+    : IFileIOServiceProvider
+{
+    int Started = 0;
+    int Stopped = 0;
+
+    TVector<std::shared_ptr<TTestFileIOService>> FileIOs;
+
+    void Start() override
+    {
+        ++Started;
+    }
+
+    void Stop() override
+    {
+        ++Stopped;
+
+        for (auto& s: FileIOs) {
+            s->Stop();
+        }
+    }
+
+    IFileIOServicePtr CreateFileIOService(TStringBuf filePath) override
+    {
+        Y_UNUSED(filePath);
+
+        auto service = std::make_shared<TTestFileIOService>();
+
+        FileIOs.push_back(service);
+
+        return service;
+    }
+};
+
+}   // namespace
+
+////////////////////////////////////////////////////////////////////////////////
+
+Y_UNIT_TEST_SUITE(TFileIOServiceProviderTest)
+{
+    Y_UNIT_TEST(ShouldCreateSingleFileIOService)
+    {
+        auto fileIO = std::make_shared<TTestFileIOService>();
+        UNIT_ASSERT_VALUES_EQUAL(0, fileIO->Started);
+
+        auto provider = CreateFileIOServiceProviderStub(fileIO);
+        provider->Start();
+
+        UNIT_ASSERT_VALUES_EQUAL(1, fileIO->Started);
+
+        auto io1 = provider->CreateFileIOService("foo");
+        UNIT_ASSERT(io1 == fileIO);
+        UNIT_ASSERT_VALUES_EQUAL(1, fileIO->Started);
+
+        auto io2 = provider->CreateFileIOService("bar");
+        UNIT_ASSERT(io2 == fileIO);
+        auto io3 = provider->CreateFileIOService("foo");
+        UNIT_ASSERT(io3 == fileIO);
+        auto io4 = provider->CreateFileIOService("buz");
+        UNIT_ASSERT(io4 == fileIO);
+
+        UNIT_ASSERT_VALUES_EQUAL(1, fileIO->Started);
+        UNIT_ASSERT_VALUES_EQUAL(0, fileIO->Stopped);
+
+        provider->Stop();
+        UNIT_ASSERT_VALUES_EQUAL(1, fileIO->Stopped);
+    }
+
+    void ShouldCreateOneServicePerOnePathImpl(ui32 pathsToServices)
```

Comment:
since this function only works for pathsToServices == 0 or pathsToServices = 1
maybe add UNIT_ASSERT(pathsToServices <= 1)